modules/dag-l0/src/main/scala/io/constellationnetwork/dag/l0/http/routes/DelegatedStakesRoutes.scala

```diff
@@ -140,6 +147,144 @@ final case class DelegatedStakesRoutes[F[_]: Async: Hasher](
       .traverse { case DelegatedStakeRecord(delegatedStaking, _, _) => DelegatedStakeReference.of(delegatedStaking) }
       .map(_.getOrElse(DelegatedStakeReference.empty))
 
+  private def toAmount(value: Long): F[Amount] =
+    if (value == 0L) Amount.empty.pure[F]
+    else
+      PosLong
+        .from(value)
+        .pure[F]
+        .map(_.leftMap(new IllegalArgumentException(_)))
+        .flatMap(Async[F].fromEither(_))
+        .map(Amount(_))
+
+  private def getRewardsInfo(
+    lastSnapshot: GlobalIncrementalSnapshot,
+    lastSnapshotInfo: GlobalSnapshotInfo
+  ): F[RewardsInfo] =
+    for {
+      emConfig <- delegatedRewardsConfig.emissionConfig
+        .get(environment)
+        .pure[F]
+        .flatMap(Async[F].fromOption(_, new RuntimeException(s"Could not retrieve emission config for env: $environment")))
+
+      (_, _, totalDelegateStake, currentTotalSupply) <- processDelegations(lastSnapshotInfo)
+      latestDelegateRewardsNoCommission <- getLatestDelegateRewardTotal(lastSnapshot, lastSnapshotInfo)
+
+      currentPrice <- toAmount(getCurrentDagPrice(emConfig))
+      nextPrice <- getNextDagPrice(emConfig)
+
+      avgRewardAmount <- calculateAverageReward(latestDelegateRewardsNoCommission, totalDelegateStake)
+      totalRewardsPerYear <- calculateAverageRewardOverAYear(avgRewardAmount, emConfig.epochsPerYear)
+
+    } yield
+      RewardsInfo(
+        epochsPerYear = emConfig.epochsPerYear,
+        currentDagPrice = currentPrice,
+        nextDagPrice = nextPrice,
+        totalDelegatedAmount = totalDelegateStake,
+        latestAverageRewardPerDag = avgRewardAmount,
+        totalDagAmount = currentTotalSupply,
+        totalRewardPerEpoch = latestDelegateRewardsNoCommission,
+        totalRewardsPerYearEstimate = totalRewardsPerYear
+      )
+
+  private def processDelegations(info: GlobalSnapshotInfo): F[(Amount, Amount, Amount, Amount)] = {
+    val activeDelegatedStakes = info.activeDelegatedStakes
+      .getOrElse(SortedMap.empty[Address, List[DelegatedStakeRecord]])
+
+    val pendingWithdrawals = info.delegatedStakesWithdrawals
+      .getOrElse(SortedMap.empty[Address, List[PendingDelegatedStakeWithdrawal]])
+
+    val totalSpendableSupply = info.balances.values.map(_.value.value).sum
+    val totalPendingSupply = pendingWithdrawals.values.flatten.map(_.rewards.value.value).sum
+
+    val (totalStakeLocked, totalActiveRewards) = activeDelegatedStakes.values.flatten.foldLeft((0L, 0L)) {
+      case ((stakeAcc, rewardsAcc), record) =>
+        val stakeAmount = record.event.value.amount.value.value
+        val rewardsAmount = record.rewards.value
+        (stakeAcc + stakeAmount, rewardsAcc + rewardsAmount)
+    }
+
+    (
+      toAmount(totalStakeLocked),
+      toAmount(totalActiveRewards),
+      toAmount(totalStakeLocked + totalActiveRewards), // totalDelegateStake
+      toAmount(totalSpendableSupply + totalPendingSupply + totalActiveRewards) // currentTotalSupply
```

Comment:
I don't think this method should be a part of this class, for instance, it's missing balances from AllowSpends. It makes way more sense to declare a method on top of the state to sum the balances.